src/main/java/gui/viewmodel/MainPageViewModel.java

```diff
@@ -0,0 +1,179 @@
+package gui.viewmodel;
+
+
+import datagateway.event.EventReader;
+import datagateway.event.ObservableEventRepository;
+import datagateway.task.ObservableTaskRepository;
+import datagateway.task.TaskReader;
+import services.eventcreation.EventInfoFromReader;
+import services.eventpresentation.CalendarEventRequestBoundary;
+import services.eventpresentation.EventInfo;
+import services.taskpresentation.TaskInfo;
+import services.taskpresentation.TaskInfoFromTaskReader;
+import services.taskpresentation.TodoListRequestBoundary;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+
+public class MainPageViewModel extends ViewModel implements Runnable{
+    private final TodoListRequestBoundary taskRepository;
+    private final CalendarEventRequestBoundary eventRepository;
+    private final List<TaskInfo> relevantTasks = new ArrayList<>();
+    private final List<EventInfo> relevantEvents = new ArrayList<>();
+    private final List<Consumer<Infos>> observers = new ArrayList<>();
+
+    public MainPageViewModel(TodoListRequestBoundary taskGetter, CalendarEventRequestBoundary eventGetter){
+        this.taskRepository = taskGetter;
+        this.eventRepository = eventGetter;
+
+        updateRelevantTasks();
+        updateRelevantEvents();
```

Comment:
Seems like these two are always paired together. Might be able to combine or wrap these methods together, which would also allow you to call the observer notify after update both lists instead of once after updating each list.